### PR TITLE
feat/30-infinite-loading-indicators

### PR DESCRIPTION
**Pull request by jeremy-babylonchain**

### CODE REVIEW

File: src/app/components/FinalityProviders/FinalityProviders.tsx [hunk 1]
```diff
@@ -1,65 +1,82 @@
 import { AiOutlineInfoCircle } from "react-icons/ai";
+import InfiniteScroll from "react-infinite-scroll-component";
 import { Tooltip } from "react-tooltip";
 
 import { FinalityProvider } from "./FinalityProvider";
 import { FinalityProvider as FinalityProviderInterface } from "@/app/types/finalityProviders";
+import { LoadingTableList, LoadingView } from "../Loading/Loading";
```
Comment (jrwbabylonchain) [inline src/app/components/FinalityProviders/FinalityProviders.tsx:7]: question. when we using `@` and `..`?  seems a bit confusing

**Comment by jeremy-babylonchain** [comment]
Delegations Desktop
https://github.com/babylonchain/simple-staking/assets/168515712/1c54cb63-e34b-4ff0-99f9-bb454e14c036

Delegations Mobile 
https://github.com/babylonchain/simple-staking/assets/168515712/f2e61ecd-8157-4ace-b7f5-12f61d963fc2

File: src/app/api/getDelegations.ts [hunk 2]
```diff
@@ -2,19 +2,16 @@ import { encode } from "url-safe-base64";
 
 import { apiWrapper } from "./apiWrapper";
 import { Delegation } from "../types/delegations";
+import { Pagination } from "../types/api";
 
 export interface PaginatedDelegations {
-  delegations: Delegation[];
-  pagination: DelegationsPagination;
+  data: Delegation[];
```
Comment (gbarkhatov) [inline src/app/api/getDelegations.ts:8]: We need to properly merge with the main not to cover up the changes that were implemented on the branch. So we don't cover @vitsalis changes

For example here it should be 
```
  delegations: Delegation[];
  pagination: DelegationsPagination;
```

File: src/app/api/getDelegations.ts [hunk 3]
```diff
@@ -2,19 +2,16 @@ import { encode } from "url-safe-base64";
...
 }
 
-export interface DelegationsPagination {
-  nextKey: string;
-}
-
-interface DelegationsAPI {
+interface DelegationsAPIResponse {
   data: DelegationAPI[];
-  pagination: PaginationAPI;
```
Comment (gbarkhatov) [inline src/app/api/getDelegations.ts:17]: same here

File: src/app/components/Delegations/Delegations.tsx [hunk 4]
```diff
@@ -322,61 +330,81 @@ export const Delegations: React.FC<DelegationsProps> = ({
...
-              modalMode === MODE_UNBOND
-                ? handleUnbondWithErrors(txID)
-                : handleWithdrawWithErrors(txID);
-            }}
-            mode={modalMode}
-          />
-        )}
-      </div>
+      {combinedDelegationsData.length === 0 ? (
+        <LoadingView />
```
Comment (gbarkhatov) [inline src/app/components/Delegations/Delegations.tsx:334]: I like the Loading View

**Review by gbarkhatov — APPROVED** [review]
Looks good to me
@jeremy-babylonchain please double check that you're no erasing previous changes from @vitsalis, @jrwbabylonchain and myself

@jrwbabylonchain once you approve this can be merged

**Review by jrwbabylonchain — APPROVED** [review]

**Pull request merged**